### PR TITLE
Remove dependency on ggpubr

### PR DESCRIPTION
**Pull request by maurolepore**
This PR removes the package `ggpubr` and makes all our CI workflows pass again. Sorry the name of the branch maybe misleading, as it mentions the package `cowplot` which this PR (used to but now) does not use.

Appart form ensuring `test()` and `check()` pass cleanly, I tested visually that the output here is similar to the output on `master` -- but @MonikaFu please check yourself, maybe by running this code from this PR and from `master`, and comparing the results. 

``` r
devtools::load_all()

data <- prepare_for_trajectory_chart(
  process_input_data(example_data),
  sector_filter = "power",
  technology_filter = "renewablescap",
  region_filter = "global",
  scenario_source_filter = "demo_2020",
  value_name = "production"
)

scenario_specs <- dplyr::tibble(
  scenario = c("sds", "sps", "cps", "worse"),
  color = c("#9CAB7C", "#FFFFCC", "#FDE291", "#E07B73"),
  label = c("SDS", "STEPS", "CPS", "worse")
)

main_line_metric <- dplyr::tibble(metric = "projected", label = "Portfolio")

additional_line_metrics <- dplyr::tibble(
  metric = "corporate_economy",
  label = "Corporate Economy"
)

p <- plot_trajectory(data,
  scenario_specs_good_to_bad = scenario_specs,
  main_line_metric = main_line_metric
)

# Not printing `p` because reprex is failing to upload the resulting plot
# p
```

<sup>Created on 2021-05-13 by the [reprex package](https://reprex.tidyverse.org) (v2.0.0)</sup>

<details style="margin-bottom:10px;">
<summary>
Standard output and standard error
</summary>

``` sh
-- nothing to show --
```

</details>﻿

Below I highlight the code chunk that determines the ralative size of the plot area relative to the legend area. I suspect @MonikaFu may want to fine tune this.

### CODE REVIEW

File: R/plot_trajectory.R [hunk 1]
```diff
@@ -260,11 +259,7 @@ add_legend <- function(plot,
 
   legend <- get_legend(p_legend)
 
-  plot <- ggarrange(
-    plot,
-    legend.grob = legend,
-    legend = "right"
-  )
+  plot <- plot_grid(plot, legend, rel_widths = c(7.5, 2.5))
```
Comment (maurolepore) [inline R/plot_trajectory.R:262]: `rel_wodths` controls the with of the `plot` relative to the `legend`.

File: R/plot_trajectory.R [hunk 2]
```diff
@@ -299,3 +290,22 @@ help_plot_area_colors <- function(data_scenarios,
...
+
+add_grobs <- function(plot, legend) {
+  gridExtra::grid.arrange(
+    gridExtra::arrangeGrob(
+      plot + theme(legend.position = "none"),
+      nrow = 1
+    ),
+    legend,
+    ncol = 2,
+    widths = c(8, 2)
```
Comment (maurolepore) [inline R/plot_trajectory.R:302]: `withds` controls the with of the `plot` relative to the `legend`.

Comment (MonikaFu) [reply R/plot_trajectory.R:302]: adjusted it in #159

**Review by MonikaFu — COMMENTED** [review]
LGTM, however this implementation still returns an object that is not an ggplot plot which is a different behaviour than for the other functions in the package.

**Review by MonikaFu — APPROVED** [review]

**Pull request merged**